src/utils/SessionLock.ts

```diff
@@ -208,13 +238,23 @@ export async function getSessionLock(onNewInstance: () => Promise<void>): Promis
     // Now add a listener for other claimants to the lock.
     window.addEventListener("storage", onStorageEvent);
 
-    // also add a listener to clear our claims when our tab closes (provided we haven't released it already)
-    window.document.addEventListener("visibilitychange", (event) => {
-        if (window.document.visibilityState === "hidden" && lockServicer !== null) {
-            prefixedLogger.info("Unloading: clearing our claims");
-            window.localStorage.removeItem(STORAGE_ITEM_PING);
-            window.localStorage.removeItem(STORAGE_ITEM_OWNER);
+    // also add a listener to clear our claims when our tab closes or navigates away
+    window.addEventListener("pagehide", (event) => {
+        // only remove the ping if we still think we're the owner. Otherwise we could be removing someone else's claim!
+        if (lockServicer !== null) {
+            prefixedLogger.info("page hide: clearing our claim");
```

Comment:
This seems noisy, should it be `debug`? (They still end up in shakes)

Reply:
yeah probably
```suggestion
            prefixedLogger.debug("page hide: clearing our claim");
```